_posts/2016-11-17-JAX-London-2016.md

```diff
@@ -0,0 +1,691 @@
+- Each running in its own process
+- Communicating with lightweight mechanisms
+- Built around business capabilities and independently deployable by fully automated deployment machinery
+- Bare minimum of centralized management
+
+Jeremy really likes Apache ActiveMQ for several reasons:
+
+- Not the fastest but the easiest to implement and maintain
+- Easy to learn
+- Great and active community
+- High Availability via a master/slave approach
+
+Finally he did a demo of an application built with ActiveMQ and Apache Camel to show how well and easy they integrate together.
+The source of the demo is available at the links below.
+
+Links:
+
+- [Presentation](https://prezi.com/07avshsn5waj/architectural-resiliency/)
+- [https://github.com/jtdeane/event-driven-microservices](https://github.com/jtdeane/event-driven-microservices)
+- [https://github.com/jtdeane/camel-standalone-router](https://github.com/jtdeane/camel-standalone-router)
+
+## Operating the Spotify backend - Niklas Gustavsson
+<img class="float-image p-image" src="{{ '/img/jax-london-2016/niklas-gustavsson.jpg' | prepend: site.baseurl }}" alt="Niklas Gustavsson" style="width: 100px"/>
+
+[Niklas Gustavsson](https://twitter.com/protocol7) immediately starts off with the culture at Spotify.
+Spotify made two very nicely animated videos that explain their culture:
+
+- [Spotify Engineering Culture Part 1](https://labs.spotify.com/2014/03/27/spotify-engineering-culture-part-1/)
+- [Spotify Engineering Culture Part 2](https://labs.spotify.com/2014/09/20/spotify-engineering-culture-part-2/)
+
+Basically it comes down to having autonomous teams called squads containing of 7-12 team members:
+
+- A squad should be fully staffed for their "mission"
+- A squad can decide how they want to tackle their issue
+- Each squad is on call for their own microservices
+
+At Spotify they have hundreds of fairly small services that only do one thing well.
+Each service is owned by a squad that implements and deploys it into production.
+Later on, they started putting ops guys in the squads in order to be able to tackle issues a lot faster.
+Jeremy also stresses that it's very important to automate as much as possible.
+
+At Spotify they have a simple incident process to avoid the same issues from reappearing:
+
+- Something explodes, an incident gets created and it gets fixed
+- (Blameless) post-mortem meeting 
+- Remediations to make sure it never happens again
+- The incident is closed
+
+System-Z is the service registry in which all services in production get registered in and the following information is available for each microservice:
+
+- Who’s on call for each service
+- When a service is down
+- What the dependencies are for each service
+- The hardware in use
+- The amount of instances
+	
+Also, in order to encourage squads to register their microservices, if you’re not registering your services in there, you’re not getting any hardware and thus you can’t get your service into production.
```

Comment:
I like the original better, I did delete one of the comma's!

Also, in order to encourage squads to register their microservices, if you’re not registering your services in there you’re not getting any hardware and thus you can’t get your service into production.